src/components/SlateEditor/plugins/mathml/EditMath.js

```diff
@@ -0,0 +1,143 @@
+import PropTypes from 'prop-types';
+import he from 'he';
+
+import EditMathModal from './EditMathModal';
+
+const mathOpenTag = '<math xmlns="http://www.w3.org/1998/Math/MathML">';
+const mathCloseTag = '</math>';
+let mathEditor;
+
+class EditMath extends Component {
+  constructor(props) {
+    super(props);
+    const { innerHTML } = props.model;
+    this.state = {
+      initialMathML: innerHTML
+        ? `${mathOpenTag}${he.decode(innerHTML)}${mathCloseTag}`
+        : undefined,
+      mathML: innerHTML
+        ? `${mathOpenTag}${he.decode(innerHTML)}${mathCloseTag}`
+        : '',
+    };
+    this.previewMath = this.previewMath.bind(this);
+    this.onHandleExit = this.onHandleExit.bind(this);
+    this.onHandleExitSave = this.onHandleExitSave.bind(this);
+    this.onHandleExitRemove = this.onHandleExitRemove.bind(this);
+    this.onCancel = this.onCancel.bind(this);
+    this.onContinue = this.onContinue.bind(this);
+  }
+
+  componentDidMount() {
+    if (window.MathJax) {
+      window.MathJax.Hub.Queue(['Typeset', window.MathJax.Hub]);
+    }
+
+    if (window.com.wiris) {
+      mathEditor = window.com.wiris.jsEditor.JsEditor.newInstance({
+        language: 'en',
+      });
+      mathEditor.setMathML(this.state.mathML);
+      mathEditor.insertInto(document.getElementById('mathEditorContainer'));
+    }
+  }
+
+  componentDidUpdate() {
+    if (window.MathJax) {
+      window.MathJax.Hub.Queue(['Typeset', window.MathJax.Hub]);
+    }
+  }
+
+  onHandleExit(closeModal) {
+    const { initialMathML, hasSaved } = this.state;
+    const mathML = he.decode(mathEditor.getMathML());
+
+    if (!hasSaved && initialMathML !== undefined && initialMathML !== mathML) {
+      this.setState({ openDiscardModal: true });
+    } else {
+      closeModal();
+    }
+  }
+
+  onHandleExitSave(closeModal) {
+    const { handleSave } = this.props;
+
+    let saveMathML = he.decode(mathEditor.getMathML());
+
+    // Removes outer tags for embed saving
+    saveMathML = saveMathML.replace(mathOpenTag, '');
+    saveMathML = saveMathML.replace(mathCloseTag, '');
+
+    handleSave(saveMathML);
+    this.setState({ hasSaved: true });
+
+    closeModal();
+  }
+
+  onHandleExitRemove(closeModal) {
+    const { onRemoveClick } = this.props;
+
+    onRemoveClick();
+    closeModal();
+  }
+
+  previewMath() {
+    const mathML = mathEditor.getMathML();
+    this.setState({ mathML });
+  }
+
+  onCancel() {
+    this.setState({ openDiscardModal: false });
+  }
+
+  onContinue() {
+    const { onExit } = this.props;
+    onExit();
+  }
+
+  render() {
+    const { mathML, openDiscardModal } = this.state;
+    const { onExit, isEditMode } = this.props;
+
+    if (!isEditMode) {
+      return null;
+    }
+
+    return (
+      <EditMathModal
+        handleExit={this.onHandleExit}
+        handleSave={this.onHandleExitSave}
+        handleRemove={this.onHandleExitRemove}
+        handleCancel={this.onCancel}
+        handleContinue={this.onContinue}
```

Comment:
Kunne du ikke bare kalt på onExit her?

Reply:
Det er litt mer tricky på grunn av pga modalen fra `frontend-packages`, for å kunne håndtere alle lukke-caser.